pkg/controllers/terminating/v1alpha1/terminate.go

```diff
@@ -38,110 +37,72 @@ type Terminator struct {
+		return true, nil
+	}
+	// 1. Get pods on node
+	pods, err := t.getPods(ctx, node.Name)
 	if err != nil {
-		return fmt.Errorf("listing draining nodes, %w", err)
+		return false, fmt.Errorf("listing pods for node %s, %w", node.Name, err)
 	}
-	// 2. Drain nodes
-	drained := []*v1.Node{}
-	for _, node := range draining {
-		// TODO: Check if Node should be drained
-		// - Disrupts PDB
-		// - Pods owned by controller object
-		// - Pod on Node can't be rescheduled elsewhere
-
-		// 2a. Get pods on node
-		pods, err := t.getPods(ctx, node)
-		if err != nil {
-			return fmt.Errorf("listing pods for node %s, %w", node.Name, err)
-		}
-		// 2b. Evict pods on node
-		empty := true
-		for _, p := range pods {
-			if !pod.IsOwnedByDaemonSet(p) {
-				empty = false
-				if err := t.coreV1Client.Pods(p.Namespace).Evict(ctx, &v1beta1.Eviction{
-					ObjectMeta: metav1.ObjectMeta{
-						Name: p.Name,
-					},
-				}); err != nil {
-					zap.S().Debugf("Continuing after failing to evict pods from node %s, %s", node.Name, err.Error())
-				}
+	// non-daemonset pods
+	empty := true
+	// 2. Evict pods on node
+	for _, p := range pods {
+		if !pod.IsOwnedByDaemonSet(p) {
+			empty = false
+			if err := t.coreV1Client.Pods(p.Namespace).Evict(ctx, &v1beta1.Eviction{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: p.Name,
+				},
+			}); err != nil {
+				zap.S().Debugf("Continuing after failing to evict pods from node %s, %s", node.Name, err.Error())
 			}
 		}
-		// 2c. If node is empty, add to list of nodes to delete
-		if empty {
-			drained = append(drained, node)
-		}
 	}
-	// 3. Delete empty nodes
-	if err := t.deleteNodes(ctx, drained); err != nil {
-		return fmt.Errorf("deleting %d nodes, %w", len(drained), err)
-	}
-	return nil
+	return empty, nil
 }
 
 // deleteNode uses a cloudprovider-specific delete to delete a set of nodes
-func (t *Terminator) deleteNodes(ctx context.Context, nodes []*v1.Node) error {
-	// 1. Delete node in cloudprovider's instanceprovider
-	if err := t.cloudprovider.TerminateFor().Terminate(ctx, nodes); err != nil {
+func (t *Terminator) terminateNode(ctx context.Context, node *v1.Node) error {
+	// 1. Delete node in cloud provider's instance provider
+	if err := t.cloudprovider.TerminateFor().Terminate(ctx, node); err != nil {
 		return fmt.Errorf("terminating cloudprovider instance, %w", err)
 	}
 	// 2. Delete node in APIServer
-	for _, node := range nodes {
-		if err := t.kubeClient.Delete(ctx, node); err != nil {
-			zap.S().Debugf("Continuing after failing to delete node %s, %s", node.Name, err.Error())
-		}
-		zap.S().Infof("Terminated node %s", node.Name)
+	if err := t.kubeClient.Delete(ctx, node); err != nil {
+		// Don't return error as Node Controller will clean this node up without underlying instance
+		zap.S().Debugf("Continuing after failing to delete node %s, %s", node.Name, err.Error())
 	}
+	zap.S().Infof("Terminated node %s", node.Name)
```

Comment:
This logline will conflict with :95

Reply:
Good catch. I had this when we were deleting multiple nodes. Removing this.